merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/services/ConstraintAction.java

```diff
@@ -51,11 +51,18 @@ public List<ConstraintResult> getViolations(final PlanId planId, final Optional<
       throws NoSuchPlanException, MissionModelService.NoSuchMissionModelException, SimulationDatasetMismatchException {
     final var plan = this.planService.getPlanForValidation(planId);
     final Optional<SimulationResultsHandle> resultsHandle$;
+    final SimulationDatasetId simDatasetId;
     if (simulationDatasetId.isPresent()) {
       resultsHandle$ = this.simulationService.get(planId, simulationDatasetId.get());
```

Comment:
I'm under the impression that resultsHandle$ will be `Optional.empty()` if the given simulation dataset id does exist, but is associated with a different plan. Is that accurate?

Reply:
That is not accurate. If the sim dataset id does exist, a `SimulationDatasetMismatchException` will be thrown. You can see it thrown on [L144 of `PostgresResultsCellRepository`](https://github.com/NASA-AMMOS/aerie/blob/refactor/adjust-constraints-error-messages/merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/PostgresResultsCellRepository.java#L144), where it bubbles all the way up to `MerlinBindings`